Userland/Utilities/sed.cpp

```diff
@@ -344,7 +344,45 @@ struct YArguments {
 
     static SedErrorOr<YArguments> parse(GenericLexer& lexer)
     {
-        return SedError::parsing_error(lexer, "not implemented"sv);
+        auto generic_error_message = "Incomplete transform command"sv;
+
+        if (lexer.is_eof())
+            return SedError::parsing_error(lexer, generic_error_message);
+
+        auto delimiter = lexer.consume();
+        if (delimiter == '\\' || delimiter == '\n')
+            return SedError::parsing_error(lexer, "\\n and \\ cannot be used as delimiters."sv);
+
+        auto characters = lexer.consume_until([is_escape_sequence = false, delimiter](char c) mutable {
```

Comment:
just `[&]` captures are fine for small non-escaping local lambdas. then you don't need the `mutable` either. (also below)

Reply:
I am declaring and initializing `is_escape_sequence` in the capture clause. It is not present in the enclosing scope and it is getting mutated in the body.

PS: New to C++ so I might be missing something.

Reply:
I haven't seen this before, but it does make sense. Thanks for explaining!

(Looks like it's used in a few other places in serenity.)